### ISSUE TITLE
Dulwich directory empty?!

### ISSUE THREAD

**Issue by jaccovanschaik** [germandutchwindtunnels/nettools#9]
When I clone the git repos and check out the dulwich branch the dulwich subdir is empty:

> git clone https://github.com/germandutchwindtunnels/nettools.git nettools
> cd nettools
> git checkout dulwich
> ll dulwich 
total 0
>

**Comment by roeles** [comment]
add --recursive flag to clone.

**Comment by jaccovanschaik** [comment]
And what's the equivalent for git pull?

**Comment by roeles** [comment]
I think you're looking for
`git submodule update --init`

**Comment by jaccovanschaik** [comment]
OK, now I'm getting this:

```
> git clone --recursive https://github.com/germandutchwindtunnels/nettools.git nettools
> cd nettools
> python AutoUpdate.py NewGui.pyw <username> <password> <switch>

Total 0 (delta 0), reused 0 (delta 0), pack-reused 0
Traceback (most recent call last):
  File "AutoUpdate.py", line 34, in <module>
    porcelain.pull(target_dir, github_url)
  File "./dulwich/dulwich/porcelain.py", line 723, in pull
    r.reset_index()
  File "./dulwich/dulwich/repo.py", line 938, in reset_index
    validate_path_element=validate_path_element)
  File "./dulwich/dulwich/index.py", line 513, in build_index_from_tree
    os.mkdir(full_path)
OSError: [Errno 17] File exists: './dulwich'

```

**Comment by roeles** [comment]
Dulwich source shows no flags regarding pulling, opening an issue at dulwich:
https://github.com/jelmer/dulwich/issues/505

**Comment by roeles** [comment]
@jaccovanschaik when I now replay your commands, I get a GUI. Can you confirm?

**Comment by jaccovanschaik** [comment]
Yep, seems to work. Can you commit to svn?

**Comment by roeles** [comment]
Done. Issue solved with commit 7f828dd9006efd6973654c19c7aad0f6cb175344
